TypeCobol/Compiler/Preprocessor/CopyWrappedAsProgramTokensDocument.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics;
+using TypeCobol.Compiler.Concurrency;
+using TypeCobol.Compiler.Directives;
+using TypeCobol.Compiler.Parser;
+using TypeCobol.Compiler.Scanner;
+using TypeCobol.Compiler.Text;
+
+namespace TypeCobol.Compiler.Preprocessor
+{
+    /// <summary>
+    /// Special ProcessedTokensDocument that wraps original lines of a copy into a fake program
+    /// to allow full parsing and analysis on copies.
+    /// </summary>
+    public class CopyWrappedAsProgramTokensDocument : ProcessedTokensDocument
+    {
+        private static ISearchableReadOnlyList<CodeElementsLine> Wrap(ISearchableReadOnlyList<CodeElementsLine> originalLines, TextSourceInfo textSourceInfo)
+        {
+            var columnsLayout = textSourceInfo.ColumnsLayout;
+            var start = columnsLayout == ColumnsLayout.CobolReferenceFormat ? 7 : 0;
+            var indent = new string(' ', start);
+
+            var lines = new ImmutableList<CodeElementsLine>.Builder(ImmutableList<CodeElementsLine>.Empty);
+
+            //Begin fake program skeleton
+            lines.Add(IdentificationDivision());
+            lines.Add(ProgramId());
```

Comment:
We also need `Decimal point is comma` by default:
```cobol
       ENVIRONMENT DIVISION.                 
       CONFIGURATION SECTION.                
       SOURCE-COMPUTER.  IBM-370.            
       SPECIAL-NAMES. DECIMAL-POINT IS COMMA.
```
This can be an option to insert or not this block of code.

Reply:
Ok to add this line in this PR, could we work on the optional/configurable aspect in another issue ?

Reply:
Well in fact I'm not sure we do need them because they won't affect program structure and `decimalPointIsComma` is set to `true` in direct copy parsing mode.

Reply:
Ok, in this case, we just need to have an unit test about that. I didn't finish to check unit tests.

Reply:
And the option to make it configurable can be done only if someone requests it.

Reply:
Done (5be38c51f6c5993d4178628570dba936c77faa3b)